qtranslate_core.php

```diff
@@ -1118,20 +1118,6 @@ function qtranxf_convertURL( $url = '', $lang = '', $forceadmin = false, $showDe
     if ( empty( $lang ) ) {
         $lang = $q_config['language'];
     }
-    if ( empty( $url ) ) {
-        // TODO refactor this hack for qtranslate-slug! We might need a hook here.
-        if ( $q_config['url_info']['doing_front_end'] && function_exists( 'qts_get_url' ) && $q_config['url_mode'] != QTX_URL_QUERY ) {
-            // quick workaround, but need a permanent solution
-            $url = qts_get_url( $lang );
-            if ( ! empty( $url ) ) {
-                if ( $showDefaultLanguage && $q_config['hide_default_language'] && $lang == $q_config['default_language'] ) {
-                    $url = qtranxf_convertURL( $url, $lang, $forceadmin, true );
-                }
```

Comment:
The intent seems: 
- `qts_get_url `does not take into account showDefaultLanguage value
- if `$q_config['hide_default_language']` is true `qts_get_url `will provide url with default language hidden
- in this case, if current language is default language, url given by `qts_get_url `is processed again in `qtranxf_convertURL `with `showDefaultLanguage=true` (no need to force in the call, I guess done to highlight the concept) to avoid url is returned without honoring `showDefaultLanguage `setting

This is superseded now as there are no shortcuts to return value before `qtranxf_get_url_for_language` call